packages/Graph/Actions/Photo/GetPhoto.cs

```diff
@@ -61,7 +61,7 @@ protected override void PopulateParameters(DialogStateManager state, Dictionary<
         {
             if (this.UserId == null)
             {
-                throw new ArgumentNullException(nameof(this.UserId));
+                throw new InvalidOperationException($"GetPhoto requires UserId property populated.");
```

Comment:
Why can't we keep the ArgumentNullException but use the two string signature to point out the param name and keep the prettier message? I just think ArgumentNullException seems like a more descriptive exception overall.

Reply:
@thomas-chung because ArgumentNullException is for arguments. There is no argument here being passed and the compiler knows it. The idea is to be consistent with the SDK and the Microsoft-wide SDK design guidelines which recommends most of these rules. Thinking it from a user's perspective, i.e. someone who is using this action, they are not invoking a method with this as argument -- they are filling a form through composer in most cases (i.e. in all composer scenarios, there is also authoring json and coded adaptive which is not the recommended code-first approach), and this exception means that they did not configure their action correctly, which feels more like a ConfigurationException, InvalidOperationException or ApplicationException.

We can choose to ignore these rules where it applies as long as it is reasonable, so it doesn't mean that they are carved in stone. In this case, I don't feel that this is one of those situations where we need to turn rules off. Instead of turning the general recommendation off, could we consider how do we change these exceptions' text so that it's nice and actionable and whether there is another type of exception that we'd prefer?

Reply:
Got it, that makes sense, thanks! In this case I think having an InvalidOperationException sounds like a good candidate.